sklearn/metrics/tests/test_score_objects.py

```diff
@@ -465,12 +468,13 @@ def test_raises_on_score_list():
     X, y = make_blobs(random_state=0)
     f1_scorer_no_average = make_scorer(f1_score, average=None)
     clf = DecisionTreeClassifier()
-    with pytest.raises(ValueError):
-        cross_val_score(clf, X, y, scoring=f1_scorer_no_average)
+    assert_warns(FitFailedWarning,
```

Comment:
we prefer `pytest.raises` construct to `assert_warns` and `assert_raises` in tests.

Reply:
Isn't the `FitFailedWarning` a bit misleading here? (unless I'm misunderstanding this)

Reply:
@adrinjalali Thanks for the comments. With the new changes, the scorer fail uses the same try/exception block as the fit function to update the score in case of failure. 

https://github.com/scikit-learn/scikit-learn/blob/77ccf19c1f5c0fd756e906e098e4ceadc964663b/sklearn/model_selection/_validation.py#L533-L552

Hence we are getting the FitFailedWarning. Do you want me to modify the name to some generic name? If so, can you suggest some?

With these changes, the test cases no longer throw ValueError, so I had a discussion in the linked issue with @jnothman and he suggested to check for the fit failed warning instead. Can you please let me know how to proceed with this? Thanks.

Reply:
I don't think we can/should change the name of the warning for when `fit` fails cause that'll be backward incompatible. Ideally we'd use either a new `ScorinFailedWarning` or just a `UserWarning` or something for when the scoring fails, i.e. in two different `try ... catch` blocks.

To test for warnings, you can do

``` python
    with warnings.catch_warnings(record=True) as warn:
        ...
    assert warn
```
to make sure no warning is raised, or to check `warn` and make sure the right warning is raised among them.

There are a few examples in the tests which you should be able to find if you look for `with warnings.catch_warnings(record=True) as`

Reply:
Done

Reply:
I have used ``UserWarning`` with a separate try exception block.